proxy.php

```diff
@@ -71,7 +72,7 @@ public function checkRequest()
 			if (!mkdir($this->cache) || !copy(dirname($this->cache) . '/index.php', $this->cache . '/index.php'))
 				return false;
 
-		if (empty($_GET['hash']) || empty($_GET['request']))
+		if (empty($_GET['hash']) || empty($_GET['request']) || ($_GET['request'] === "http:") || ($_GET['request'] === "https:"))
```

Comment:
^ What are these cases for?

Reply:
see https://www.simplemachines.org/community/index.php?topic=558395.msg3962226#msg3962226

Reply:
Rather than testing those specific cases, it would be better to test if `$_GET['request']` is a valid URL. This should do it:
```
$_GET['request'] = filter_var($_GET['request'], FILTER_VALIDATE_URL);
if (empty($_GET['hash'] || empty($_GET['request']))
	return false;
```

Note: Unfortunately, `filter_var($url, FILTER_VALIDATE_URL)` doesn't work for URLs that contain non-ASCII characters, so this will fail for an image hosted at, e.g., http://www.exämple.com/image.jpg. But I am just finishing putting together a PR to fix that. So for now go ahead and use `$_GET['request'] = filter_var($_GET['request'], FILTER_VALIDATE_URL);`, and then I will replace that line once this is merged.

Reply:
It's not the intention here to test the url,
only to catch some cases where the stuff is empty,
when the url is "http:" or "https:" means that it got nothing.

Reply:
`filter_var($_GET['request'], FILTER_VALIDATE_URL)` will return `false` if `$_GET['request']` is not a valid URL. Then the `empty($_GET['request'])` will be triggered in the if statement.

Reply:
You are correct that the existing code didn't check the URL properly, but it should have. The fact it wasn't being checked properly is the reason why you had to add ` || ($_GET['request'] === "http:") || ($_GET['request'] === "https:")` to the if statement. If the URL was being properly checked, adding those cases would not have been necessary to begin with.